examples/user_guide/17-Dashboards.ipynb

```diff
@@ -117,15 +117,45 @@
    "outputs": [],
    "source": [
     "# Apply rolling mean\n",
-    "window = Params(explorer, ['rolling_window'])\n",
-    "smoothed = rolling(stock_dmap, streams=[window])\n",
+    "smoothed = rolling(stock_dmap, rolling_window=explorer.param.rolling_window)\n",
     "\n",
     "# Find outliers\n",
-    "outliers = rolling_outlier_std(stock_dmap, streams=[window]).opts(\n",
-    "    hv.opts.Scatter(color='red', marker='triangle')\n",
-    ")\n",
+    "outliers = rolling_outlier_std(stock_dmap, rolling_window=explorer.param.rolling_window).opts(\n",
+    "    color='red', marker='triangle')\n",
+    "\n",
+    "pn.Row(explorer.param, (smoothed * outliers).opts(width=600, padding=0.1))"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## A function based approach\n",
+    "\n",
+    "Instead of defining a whole Parameterized class we can also use the ``depends`` decorator to directly link the widgets to a DynamicMap callback function. This approach makes the link between the widgets and the computation very explicit at the cost of tying the widget and display code very closely together.\n",
+    "\n",
+    "Instead of declaring the dependencies as strings we map the parameter instance to a particular keyword argument in the ``depends`` call. In this way we can link the symbol to the ``RadioButtonGroup`` value and the ``variable`` to the ``Select`` widget value:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "symbol = pn.widgets.RadioButtonGroup(options=stock_symbols)\n",
+    "variable = pn.widgets.Select(options=variables)\n",
+    "window = pn.widgets.IntSlider(name='Rolling Window', value=10, start=1, end=365)\n",
+    "\n",
+    "@pn.depends(symbol=symbol.param.value, variable=variable.param.value)\n",
+    "def load_symbol_cb(symbol, variable):\n",
+    "    return load_symbol(symbol, variable)\n",
+    "\n",
+    "dmap = hv.DynamicMap(load_symbol_cb)\n",
+    "\n",
+    "smoothed = rolling(stock_dmap, rolling_window=window.param.value)\n",
     "\n",
-    "pn.Row(explorer.param, (smoothed * outliers).opts(width=600))"
+    "pn.Row(pn.WidgetBox('## Stock Explorer', symbol, variable, window), smoothed.opts(width=500))"
    ]
```

Comment:
I am not sure that it is worth the effort to change, but I think that if we were starting this notebook from scratch we would introduce the conceptually simpler function approach *first* and then talk about classes and why you want to use them. Even if we decide to leave things in this order, I think a paragraph is necessary here to explain why you might want to use classes at all. Any thoughts @jbednar as to whether it is worth reworking the order in this PR?

Reply:
This sentence above was my attempt at explaining the difference:

>This approach makes the link between the widgets and the computation very explicit at the cost of tying the widget and display code very closely together.

You're right that it probably deserves a larger discussion though.

Reply:
I agree with reordering it, but that doesn't have to hold up this PR.